pkg/cloudprovider/providers/azure/azure_storageaccount.go

```diff
@@ -37,20 +37,17 @@ func (az *Cloud) getStorageAccounts(matchingAccountType string) ([]accountWithLo
 
 	accounts := []accountWithLocation{}
 	for _, acct := range *result.Value {
-		if acct.Name != nil {
-			name := *acct.Name
-			loc := ""
-			if acct.Location != nil {
-				loc = *acct.Location
+		if acct.Name != nil && acct.Location != nil && acct.Sku != nil {
+			storageType := string((*acct.Sku).Name)
+			if matchingAccountType != "" && !strings.EqualFold(matchingAccountType, storageType) {
+				continue
 			}
-			storageType := ""
-			if acct.Sku != nil {
-				storageType = string((*acct.Sku).Name)
-				if matchingAccountType != "" && !strings.EqualFold(matchingAccountType, storageType) {
-					continue
-				}
+
+			location := *acct.Location
+			if location != "" && !strings.EqualFold(matchingLocation, location) {
+				continue
 			}
-			accounts = append(accounts, accountWithLocation{Name: name, StorageType: storageType, Location: loc})
+			accounts = append(accounts, accountWithLocation{Name: *acct.Name, StorageType: storageType, Location: location})
```

Comment:
We'd better use the actual type & location from acct object here, since storageType and location could be empty

Reply:
`storageType`, `Location` is from acc object.
```
			storageType := string((*acct.Sku).Name)
			if matchingAccountType != "" && !strings.EqualFold(matchingAccountType, storageType) {
				continue
			}

			location := *acct.Location
			if matchingLocation != "" && !strings.EqualFold(matchingLocation, location) {
				continue
			}
```